src/summary.py

```
@@ -223,161 +228,410 @@ async def unreadsummary(
 
-    # Grab variables from database
     user, server = setup_user(str(ctx.author), ctx.guild.name, ctx.guild.id)
-    language = user["language"]
-    if mode in user["modes"]:
-        mode_prompt = user["modes"][mode]
-    else:
+    if not validate_mode(ctx, mode, user):
+        logging.info(f"Invalid mode: {mode}")
+        return
+
+    use_a_thread = use_a_thread(ctx, user, secret_mode)
```

Ruff (0.8.2)

Local variable `use_a_thread` referenced before assignment

(F823)